_shared/local_extensions.py

```diff
@@ -110,11 +113,74 @@ def include_exists(self, context, path):
     def include(self, context, path, indent=0):
         """Return the lines from the project's .cookiecutter/includes/{path} or an empty string."""
         try:
-            lines = self._readlines(context, path)
+            file_ = self._open(context, path)
         except (FileNotFoundError, NotADirectoryError):
             return ""
 
-        return textwrap.indent("".join(lines), " " * indent)
+        return textwrap.indent("".join(file_.readlines()), " " * indent)
+
+    @pass_context
+    def include_json(self, context, path):
+        """Return the project's .cookiecutter/includes/{path} data or None.
+
+        Return the contents of the project's .cookiecutter/includes/{path} JSON
+        file deserialized into a Python object.
+
+        Return None if the project has no .cookiecutter/includes/{path} file.
+        """
+        return json.load(self._open(context, path))
+
+    @pass_context
+    def supervisor_settings(self, context):
+        """Return the settings from the project's supervisor.json file.
+
+        Or return the default supervisor settings.
+        """
+        try:
+            settings = self.include_json(context, "supervisor.json")
+        except FileNotFoundError:
+            settings = {}
+
+        if not settings.get("programs"):
+            if context["cookiecutter"]["_directory"] == "pyramid-app":
+                settings["programs"] = {
+                    "web": {
+                        "settings": {
+                            "command": f"gunicorn --bind :{context['cookiecutter']['port']} --workers 1 --reload --timeout 0 --paste conf/development.ini",
+                        }
+                    }
+                }
+            else:
+                settings["programs"] = {
+                    "app": {
+                        "settings": {
+                            "command": "python3 -m {context['cookiecutter']['package_name']}",
+                        }
+                    }
+                }
```

Comment:
Newly-generated non-web apps get a process that just run's the package's `__main__.py` file.